From 8a64531c2d4ae5c3217d3fdca5d2a439ca7893b9 Mon Sep 17 00:00:00 2001
From: Teng Long <dyroneteng@gmail.com>
Date: Tue, 15 Nov 2022 14:13:48 +0800
Subject: [RFC PATCH 3/6] ls-tree: optimize params of
 'show_tree_common_default_long()'

Function 'show_tree_common_default_long' has 3 parameters:

  * struct strbuf *base
  * const char *pathname
  * const size_t baselen

And the struct 'show_tree_data' which was introduced in
'e81517155e0370ae5433d256046ab287bb10d04d' is:

  struct show_tree_data {
	       unsigned mode;
	       enum object_type type;
         const struct object_id *oid;
         const char *pathname;
         struct strbuf *base;
  };

Actually, the struct includes all the data that we need to pass to the
function, so we could make a small refactoring by using 'show_tree_data'
struct directly.

Signed-off-by: Teng Long <dyroneteng@gmail.com>
---
 builtin/ls-tree.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/builtin/ls-tree.c b/builtin/ls-tree.c
index 8256fc0bc97..afb65af4280 100644
--- a/builtin/ls-tree.c
+++ b/builtin/ls-tree.c
@@ -200,15 +200,15 @@ static int show_tree_common(struct show_tree_data *data, int *recurse,
 	return ret;
 }
 
-static void show_tree_common_default_long(struct strbuf *base,
-					  const char *pathname,
-					  const size_t baselen)
+static void show_tree_common_default_long(struct show_tree_data *data)
 {
-	strbuf_addstr(base, pathname);
-	write_name_quoted_relative(base->buf,
+	int base_len = data->base->len;
+
+	strbuf_addstr(data->base, data->pathname);
+	write_name_quoted_relative(data->base->buf,
 				   chomp_prefix ? ls_tree_prefix : NULL, stdout,
 				   line_termination);
-	strbuf_setlen(base, baselen);
+	strbuf_setlen(data->base, base_len);
 }
 
 static int show_tree_default(const struct object_id *oid, struct strbuf *base,
@@ -225,7 +225,7 @@ static int show_tree_default(const struct object_id *oid, struct strbuf *base,
 
 	printf("%06o %s %s\t", data.mode, type_name(data.type),
 	       find_unique_abbrev(data.oid, abbrev));
-	show_tree_common_default_long(base, pathname, data.base->len);
+	show_tree_common_default_long(&data);
 	return recurse;
 }
 
@@ -255,7 +255,7 @@ static int show_tree_long(const struct object_id *oid, struct strbuf *base,
 
 	printf("%06o %s %s %7s\t", data.mode, type_name(data.type),
 	       find_unique_abbrev(data.oid, abbrev), size_text);
-	show_tree_common_default_long(base, pathname, data.base->len);
+	show_tree_common_default_long(&data);
 	return recurse;
 }
 
-- 
2.38.1.426.g770fc8806cb.dirty

